src/Commands/core/ArchiveRestoreCommands.php

```diff
@@ -0,0 +1,611 @@
+                null,
+                null,
+                $options['code-composer-install-timeout']
+            );
+            $process->mustRun();
+        }
+    }
+
+    /**
+     * Imports Drupal files to the site.
+     *
+     * @param string $source
+     *   The path to the source directory.
+     * @param array $options
+     *   The options.
+     *
+     * @throws \Exception
+     */
+    protected function importFiles(string $source, array $options): void
+    {
+        $this->logger()->info('Importing files...');
+
+        if (!is_dir($source)) {
+            throw new Exception(dt('The source directory !path not found for files.', ['!path' => $source]));
+        }
+
+        $destinationAbsolute = $this->fileImportAbsolutePath($options['files-destination-relative-path']);
+
+        if (
+            is_dir($destinationAbsolute) &&
+            (!$options['code'] || !$options['overwrite']) &&
+            !$this->io()->confirm(
+                dt(
+                    'Destination Drupal files path !path already exists. Are you sure you want restore Drupal files archive into it?',
+                    [
+                        '!path' => $destinationAbsolute,
+                    ]
+                )
+            )
+        ) {
+            throw new UserAbortException();
+        }
+
+        $this->filesystem->mkdir($destinationAbsolute);
+        $this->rsyncFiles($source, $destinationAbsolute);
+    }
+
+    /**
+     * Determines the path where files should be extracted.
+     *
+     * @param null|string $destinationRelative
+     *   The relative path to the Drupal files directory.
+     *
+     * @return string
+     *   The absolute path to the Drupal files directory.
+     *
+     * @throws \Exception
+     */
+    protected function fileImportAbsolutePath(?string $destinationRelative): string
+    {
+        // If the user specified the path to the files directory, use that.
+        if ($destinationRelative) {
+            return Path::join($this->getDestinationPath(), $destinationRelative);
+        }
+
+        // If we are extracting over an existing site, query Drupal to get the files path
+        $bootstrapManager = Drush::bootstrapManager();
+        $path = $bootstrapManager->getComposerRoot();
+        if (!empty($path)) {
+            try {
+                $bootstrapManager->doBootstrap(DrupalBootLevels::FULL);
+                return Drupal::service('file_system')->realpath('public://');
+            } catch (Throwable $t) {
+                $this->logger()->warning('Could not bootstrap Drupal site at destination to determine file path');
+            }
+        }
+
+        // Find the Drupal root for the archived code, and assume sites/default/files.
+        $composerRoot = $this->getDestinationPath();
+        $drupalFinder = new DrupalFinder();
+        if ($drupalFinder->locateRoot($composerRoot)) {
+            return Path::join($drupalFinder->getDrupalRoot(), 'sites/default/files');
```

Comment:
Here's an existing example where we instantiate a DrupalFinder instance to find the Drupal Root.